pyblish/plugin.py

```diff
@@ -780,6 +782,46 @@ def current_host():
     return pyblish._registered_hosts[-1] or "unknown"
 
 
+def register_callback(signal, callback):
+
+    if signal in pyblish._registered_callbacks:
+        pyblish._registered_callbacks[signal].append(callback)
+    else:
+        pyblish._registered_callbacks[signal] = [callback]
+
+
+def deregister_callback(callback):
+
+    for signal in pyblish._registered_callbacks:
+        if callback in pyblish._registered_callbacks[signal]:
+            pyblish._registered_callbacks[signal].remove(callback)
+
+
+def deregister_callbacks(signal):
+
+    pyblish._registered_callbacks.pop(signal, None)
+
+
+def deregister_all_callbacks():
+
+    pyblish._registered_callbacks = dict()
+
+
+def registered_callbacks():
+
+    return pyblish._registered_callbacks
+
+
+def emit(signal, **kwargs):
+
+    if signal in pyblish._registered_callbacks:
```

Comment:
Ah.. I can't tell whether it'd make a difference. Maybe it will, but let's leave it for when it becomes obvious.